js-envs/src/main/scala/org/scalajs/jsenv/nodejs/AbstractNodeJSEnv.scala

```diff
@@ -134,12 +134,37 @@ abstract class AbstractNodeJSEnv(
      */
     override protected def writeJSFile(file: VirtualJSFile,
         writer: Writer): Unit = {
+
+      def writeImport(file: File): Unit = {
+        val uri = file.toURI.toASCIIString
+        val importerFile = new MemVirtualJSFile("importer.js")
+        importerFile.content = {
+          s"""
+            |import("${escapeJS(uri)}").catch(e => {
+            |  /* Make sure to fail the process, but give time to Node.js to
+            |   * display a good stack trace before that.
+            |   */
+            |  setTimeout(() => process.exit(1), 100);
+            |  throw e;
+            |});
+          """.stripMargin
```

Comment:
Hum, how is this supposed to work? There is absolutely no guarantee that once this file is required, anything is loaded...

So if there is code like this:
```javascript
require("foo_importer.js");
foo.blah();
```

The second line might randomly fail because `foo` is not loaded yet.

Reply:
Hmmm... I see; ok I guess for the 0.6 series with it's unclear semantics of what files passed to a `JSEnv` mean this is OK.